includes/wikia/models/DesignSystemGlobalNavigationModelV2.class.php

```diff
@@ -92,11 +92,64 @@ private function getCreateWiki( string $trackingLabel ) {
 		];
 	}
 
-	private function getHref( $hrefKey, $protocolRelative = false ) {
+	protected function getDomainForCityId( $cityId ) {
+		global $wgServer;
+		if ( $cityId == null ) {
+			return $wgServer;
+		} else {
+			return  WikiFactory::cityIDtoUrl( $cityId );
+		}
+	}
+
+	protected function getDevDomain() {
+		global $wgDevDomain, $wgDevelEnvironment;
+		if ( !$wgDevelEnvironment || empty( $wgDevDomain ) ) {
+			return false;
+		}
+
+		return $wgDevDomain;
+	}
+
+	public function usePrimaryDomainInUrl( $url, $cityId = null ) {
```

Comment:
This name isn't really telling what's happening and the method is complex enough to be hard to follow. How about adding PHPDoc with explanation and some examples?

Reply:
Update in 9062143475d